modules/tinymce/src/themes/silver/main/ts/ui/menus/menu/MenuConversion.ts

```diff
@@ -66,39 +74,35 @@ const getFromExpandingItem = (item: Menu.NestedMenuItemSpec, menuItems: MenuItem
   };
 };
 
-const getFromItem = (item: SingleMenuItemSpec, menuItems: MenuItemRegistry) => isExpandingMenuItem(item) ? getFromExpandingItem(item, menuItems) : {
-  item,
-  menus: { },
-  expansions: { }
-};
-
-const generateValueIfRequired = (item: SingleMenuItemSpec): SingleMenuItemSpec => {
-  // Separators don't have a value, so just return the item
-  if (isSeparator(item)) {
-    return item;
-  } else {
-    // Use the value already in item if it has one.
-    const itemValue = Obj.get<any, string>(item, 'value').getOrThunk(() => Id.generate('generated-menu-item'));
-    return Merger.deepMerge({ value: itemValue }, item);
-  }
+const generateValueIfRequired = (item: Menu.NestedMenuItemSpec): Menu.NestedMenuItemSpec & { value: string } => {
+  // Use the value already in item if it has one.
+  const itemValue = Obj.get<any, string>(item, 'value').getOrThunk(() => Id.generate('generated-menu-item'));
+  return Merger.deepMerge({ value: itemValue }, item);
 };
 
 // Takes items, and consolidates them into its return value
-const expand = (items: string | Array<string | SingleMenuItemSpec>, menuItems: MenuItemRegistry) => {
+const expand = (items: string | Array<string | SingleMenuItemSpec>, menuItems: MenuItemRegistry): ExpandedMenus => {
```

Comment:
This is a bit of a rewrite, as instead of trying to do the expand logic for every item type, we now only do it for things that can expand (e.g. nested menus). The rest we just add the item to the list and move on to processing the next item.